### ISSUE TITLE
source description has inconvenient indentation

### ISSUE THREAD

**Issue by debralacoste** [DDMAL/CantusDB#1972]
On the first line of the "description" field, there seems to be auto-indentation. We would like this removed and be able to place the text fully left-justified if we wish. This may be related to the issue #1239 where there are additional spaces between paragraphs (which we also want removed). Basically, we would like to use the usual html coding as required within the text blocks of the source descriptions. If there is a global change to the way these are displayed, no worries! We will go into each one and add the appropriate html tags to make the texts look ok. Give us a heads up when you're going to make this change (assuming it's the CSS?) so we have staff ready to react.

**Comment by debralacoste** [comment]
Ok, Issue #1239 has a Loooong discussion where there is the suggestion to use markdown instead of HTML. Note that we are removing bibliographies from the Cantus Database source descriptions (and replacing them with our new Zotero archive) so these text blocks do not need to be considered. The existing tables in a few of the descriptions could be converted to images. Does that help us? Explain markdown to me, and maybe that's our way forward!

**Comment by ahankinson** [comment]
Markdown is a simple text format that lets you write some minimal structure and a markdown processor turns it into HTML. 

So for example, for section headings you use the hash / pound symbol to write different levels of heading:

```
# Level one heading
## Level two heading
### Level three heading
```

It also lets you easily write links to text using a combination of the `[]` and `()` characters:

```
The [text inside square brackets](https://google.com) will be linked to Google.com
```

GitHub comments like the one you are writing supports markdown so that text should appear as:

The [text inside square brackets](https://google.com) will be linked to Google.com

In fact, if you click the little `Reference` Icon in the top-right corner of the GitHub comment box, it gives you a reference sheet for the supported methods of "marking up" a text with markdown. 

I like it because I find it easier to teach someone how to write minimal text to add simple things like links or lists, than I do trying to troubleshoot why a complex "Word-like" HTML editor is producing really crummy HTML and messing up the page layout. Since it's a minimal format, you can also restrict people from doing silly things like changing the font or messing with line spacing.

**Comment by debralacoste** [comment]
Thanks Andrew!
Two further questions for the whole team.
1) If we implement markdown within the "Summary" and "Description" fields, will we be able to access the HTML if we want to edit it directly for some reason?
2) Will we be able to control the display? I ask this, because the views between "Preview" and actual for our descriptions do not currently match. Will changing the coding method fix this?
See screenshots: 1st is "edit" view, 2nd is "preview" within the data-entry screen, and 3rd is what shows on the site after saving.

EDIT:
<img width="870" height="330" alt="Image" src="https://github.com/user-attachments/assets/b44c6ccb-4269-4836-aa68-893c4bef17dc" />

PREVIEW:
<img width="873" height="340" alt="Image" src="https://github.com/user-attachments/assets/322e0160-5174-4b46-ae13-8cf420b4a1d4" />

PUBLIC:
<img width="864" height="271" alt="Image" src="https://github.com/user-attachments/assets/a3ba48d6-498a-4af6-8803-c3583393bb8d" />

If I could choose, I would like "PREVIEW" and "PUBLIC" to be the same.

**Comment by debralacoste** [comment]
One further comment: I had told Dylan a couple of years ago that we musicologists could certainly handle a tiny bit of HTML code. Through the numerous rebuilds and Drupal versions etc. there have been many 'rich-text' fields and other contaminants. When I update the DACT app, for instance, I copy the text to a text editor - .txt - first, and then I copy it from there to the app, and that minimizes the junk that Word sometimes adds. If markdown is going to convert to HTML anyways, we could skip that step and just have an open field that we (staff) monitor for formatting. (I expect that we'll need to redo all the description fields, no matter which way we go, so we can discuss the workflow required.)

**Comment by debralacoste** [comment]
Also, just a quick note that in my screenshots (above), I originally had more "br"s but I took them out while I was experimenting with views. (noticing now that I cannot write angle bracket-br-angle bracket in this comment, because it changes it to a line break - yay markdown doing its job!)

**Comment by ahankinson** [comment]
> If we implement markdown within the "Summary" and "Description" fields, will we be able to access the HTML if we want to edit it directly for some reason?

Generally, no. The Markdown gets stored as the text in the database, and then rendered according to styles and rules when the user loads the page. Thus there is no "HTML" to edit -- the HTML gets generated on the fly when a page loads. 

That said, you can modify the styling of the rendering by adjusting the page's stylesheet, so it can be customized to fit your page. But this is done by the developer and then applied to the generated HTML. 

> Will we be able to control the display? I ask this, because the views between "Preview" and actual for our descriptions do not currently match. Will changing the coding method fix this?
See screenshots: 1st is "edit" view, 2nd is "preview" within the data-entry screen, and 3rd is what shows on the site after saving.

The "Preview" and "Public" here are not using the same stylesheet. There is a 2em text indent being applied to the stylesheet.

Actually, looking at the code of the rendered page, I think Markdown is already being applied here. So you shouldn't write HTML in these boxes!


NB: In these GitHub issues, it's super helpful if you can provide links to pages so that we can see what you're seeing. It took me a few minutes to find this one: https://cantusdatabase.org/source/1000458

**Comment by ahankinson** [comment]
> I cannot write angle bracket-br-angle bracket in this comment

You can if you enclose them in back ticks to apply "code" formatting: `<br>`

**Comment by ahankinson** [comment]
PS: I edited your source description to experiment with the markdown, so that's why the `<p>`s have gone missing from them.

**Comment by debralacoste** [comment]
Will do - re: links! I know you guys prefer that, and this started out being a more generic "it's always indented" comment. I'm going to look up what "back ticks" are now. No worries about editing anything - this one is not published yet. It's Jennifer's fragment, and I'm here in Halifax so we're indexing it. :)

**Comment by ahankinson** [comment]
Even with generic "it's always indented" comments, having somebody reference a concrete instance where you know it's a problem to see if the problem has been fixed is quite useful.

**Comment by ahankinson** [comment]
You mean "Mama Bear" isn't an official shelfmark?

**Comment by debralacoste** [comment]
Papa Bear and Baby Bear are lined up against the stairwell. :)

**Comment by debralacoste** [comment]
also related to #1957
